src/EntryPoints/WikibaseFacetedSearchHooks.php

```diff
@@ -224,4 +241,146 @@ public static function onCirrusSearchAddQueryFeatures( SearchConfig $config, arr
+
+		if ( $config->linkTargetSitelinkSiteId === null ) {
+			$entityId = WikibaseRepo::getEntityIdLookup()->getEntityIdForTitle( $page->getTitle() );
+
+			if ( $entityId !== null ) {
+				$fields = array_merge( $fields, self::getFieldValues( $entityId ) );
+			}
+
+			return;
+		}
+
+		if ( !( $handler instanceof WikitextContentHandler ) ) {
+			return;
+		}
+
+		try {
+			$entityId = WikibaseRepo::getStore()->newSiteLinkStore()->getEntityIdForLinkedTitle(
+				$config->linkTargetSitelinkSiteId,
+				$page->getTitle()->getText()
+			);
+		} catch ( Exception $exception ) {
+			return;
+		}
+
+//		print_r("\nentityId:$entityId");
+
+		if ( $entityId === null ) {
+			return;
+		}
+
+		$fields = array_merge( $fields, self::getFieldValues( $entityId ) );
+	}
+
+	private static function getFieldValues( EntityId $entityId ): array {
+		$entity = WikibaseRepo::getEntityLookup()->getEntity( $entityId );
+
+		if ( !( $entity instanceof Item ) ) {
+			return [];
+		}
+//		print_r("\nITEM\n");
+
+		$config = WikibaseFacetedSearchExtension::getInstance()->getConfig();
+
+		$statements = $entity->getStatements();
+
+		$instanceTypeId = $statements->getByPropertyId( $config->getInstanceOfId() );
+		$instanceTypeSnak = $instanceTypeId->getMainSnaks()[0] ?? null;
+
+		if ( !( $instanceTypeSnak instanceof PropertyValueSnak ) ) {
+			return [];
+		}
+
+		$instanceTypeValue = $instanceTypeSnak->getDataValue();
+
+//		print_r('$instanceTypeValue');
+
+		if ( !( $instanceTypeValue instanceof EntityIdValue ) ) {
+			return [];
+		}
+
+		$properties = array_values(
+			array_map(
+				fn( FacetConfig $config ) => $config->propertyId,
+				$config->getFacetConfigForInstanceType( $instanceTypeValue->getEntityId() )
+			)
+		);
+
+		$values = [];
+
+		foreach ( $properties as $property ) {
+			$propertyStatements = $statements->getByPropertyId( $property );
+
+			if ( $propertyStatements->count() === 0 ) {
+				continue;
+			}
+
+			$name = 'wbfs_' . $property->getSerialization();
+
+			foreach( $propertyStatements->toArray() as $statement ) {
+				$values[$name][] = self::getSnakValue( $statement->getMainSnak() );
+			}
+
+			if ( count( $values[$name] ) === 1 ) {
+				$values[$name] = $values[$name][0];
+			}
+		}
+
+//		print("\n\nVALUES:");
+//		print_r( $values );
+//		print("\n");
+
+		return $values;
+	}
+
+	private static function getSnakValue( Snak $snak ): mixed {
+		if ( !( $snak instanceof PropertyValueSnak ) ) {
+			return null;
+		}
+
+		$dataValue = $snak->getDataValue();
+
+		if ( $dataValue instanceof QuantityValue ) {
+			return $dataValue->getAmount()->getValueFloat();
+		}
+
+		if ( $dataValue instanceof StringValue ) {
+			return $dataValue->getValue();
+		}
+
+		if ( $dataValue instanceof TimeValue ) {
+			return ltrim( $dataValue->getTime(), '+-' );
+		}
+
+		if ( $dataValue instanceof EntityIdValue ) {
+			return $dataValue->getEntityId()->getSerialization();
+		}
+
+		return null;
+	}
```

Comment:
Good candidate for own service

Similar code in Semantic Wikibase: https://github.com/ProfessionalWiki/SemanticWikibase/blob/master/src/Translation/DataValueTranslator.php#L26